hash.c

```diff
@@ -4160,13 +4160,37 @@ rb_hash_update_block_i(VALUE key, VALUE value, VALUE hash)
  *    update(*other_hashes) -> self
  *    update(*other_hashes) { |key, old_value, new_value| ... } -> self
  *
- *  Like Hash#merge, but modifies and returns +self+ instead of a new hash:
+ *  Updates values and/or adds entries to +self+; returns +self+.
  *
- *    season = {AB: 75, H: 20, HR: 3, SO: 17, W: 11, HBP: 3}
- *    today = {AB: 3, H: 1, W: 1}
- *    yesterday = {AB: 4, H: 2, HR: 1}
- *    season.update(yesterday, today) {|key, old_value, new_value| old_value + new_value }
- *    # => {AB: 82, H: 23, HR: 4, SO: 17, W: 12, HBP: 3}
+ *  Each argument +other_hash+ in +other_hashes+ must be a hash.
+ *
+ *  With no block given, for each successive entry +key+/+new_value+ in each successive +other_hash+:
+ *
+ *  - If +key+ is in +self+, sets <tt>self[key] = new_value</tt>, whose position is unchanged:
+ *
+ *      h.update({bat: 3, foo: -1}, {bat: 4, bar: -1}) # => {foo: -1, bar: -1, baz: 2, bat: 4}
+ *
+ *  - If +key+ is not in +self+, adds the entry at the end of +self+:
+ *
+ *      h = {foo: 0, bar: 1, baz: 2} # => {foo: 0, bar: 1, baz: 2}
```

Comment:
These examples are not right. `h` is not defined in the first example and this example doesn't demonstrate anything.

Reply:
Fixed.